gravitee-apim-console-webui/src/management/api/api-traffic-v4/analytics/components/api-analytics-proxy-filter-bar/api-analytics-proxy-filter-bar.component.ts

```diff
@@ -108,148 +82,106 @@ export class ApiAnalyticsProxyFilterBarComponent implements OnInit {
-    const filteredList = (currentList || []).filter((item) => item !== value);
-    formControl.setValue(filteredList.length > 0 ? filteredList : null);
+  private createProxySpecificFormControls(): ProxySpecificForm {
+    return {
+      [FILTER_KEYS.HTTP_STATUSES]: this.formBuilder.control<string[] | null>(null),
+      [FILTER_KEYS.APPLICATIONS]: this.formBuilder.control<string[] | null>(null),
+    };
   }
 
-  removeFilter(key: string, value: string) {
-    if (key === 'httpStatuses') {
-      this.removeValueFromFilter(this.form.controls.httpStatuses.value, value, this.form.controls.httpStatuses);
-    } else if (key === 'plans') {
-      this.removeValueFromFilter(this.form.controls.plans.value, value, this.form.controls.plans);
-    } else if (key === 'applications') {
-      this.removeValueFromFilter(this.form.controls.applications.value, value, this.form.controls.applications);
-    }
+  private setupProxySpecificFormSubscriptions(): void {
+    this.form
+      .get(FILTER_KEYS.HTTP_STATUSES)
+      ?.valueChanges.pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe((httpStatuses: any) => {
+        this.emitFilters({ [FILTER_KEYS.HTTP_STATUSES]: httpStatuses });
+      });
+
+    this.form
+      .get(FILTER_KEYS.APPLICATIONS)
+      ?.valueChanges.pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe((applications: any) => {
+        this.emitFilters({ [FILTER_KEYS.APPLICATIONS]: applications });
+      });
   }
 
-  resetAllFilters() {
-    this.emitFilters({ httpStatuses: null, plans: null, applications: null });
+  protected override getSpecificFormData(filters: ProxyFilters): any {
+    return {
+      [FILTER_KEYS.HTTP_STATUSES]: filters[FILTER_KEYS.HTTP_STATUSES],
+      [FILTER_KEYS.APPLICATIONS]: filters[FILTER_KEYS.APPLICATIONS],
+    };
   }
 
-  private updateFormFromFilters(filters: ApiAnalyticsProxyFilters) {
-    if (this.form) {
-      this.form.patchValue({
-        timeframe: {
-          period: filters.period,
-          from: filters.from ? moment(filters.from) : null,
-          to: filters.to ? moment(filters.to) : null,
-        },
-        plans: filters.plans,
-        httpStatuses: filters.httpStatuses,
-        applications: filters.applications,
-      });
-    }
+  protected override getSpecificFilterReset(): Partial<ProxyFilters> {
+    return {
+      [FILTER_KEYS.HTTP_STATUSES]: null,
+      [FILTER_KEYS.APPLICATIONS]: null,
+    };
   }
 
-  private emitFilters(partial: Partial<ApiAnalyticsProxyFilters>) {
-    this.filtersChange.emit({
-      ...this.activeFilters(),
-      ...partial,
-    });
+  protected override removeSpecificFilter(key: string, value: string): void {
+    if (key === FILTER_KEYS.HTTP_STATUSES) {
+      const control = this.form.get(FILTER_KEYS.HTTP_STATUSES) as FormControl<string[] | null>;
+      this.removeValueFromFilter(control.value, value, control);
```

Comment:
```suggestion
      this.removeValueFromFilter(this.form.controls.httpStatuses, value);
```
The signature of `removeValueFromFilter` can be simplified I think 🤔 
And by accessing the control via `.controls`, it will never be null.